### PR TITLE
Remove Lodash dependency

### PR DESCRIPTION
**Pull request by brandur**
Lodash is being used in relatively few places and as described in #259
can cause some problems in transforming objects to versions that cannot
be properly encoded.

Here we remove Lodash. Much of its functionality has been added to ES6
anyway, so we can eventually even remove some of the vendored functions
that we have.

Fixes a report received in #259.

### CODE REVIEW

**Comment by brandur** [comment]
@floatingLomas @SlexAxton Hey! One thing that we used from Lodash was a function that would allow us to check whether an object was "plain" (i.e. just one with basic key/value pairs) which seems to be useful when checking certain function parameters so that we can ensure that nothing crazy gets passed in. Here I ended up vendoring this method into `utils` because it seems like something that couldn't hurt for us to keep doing. I'll be the first to admit that I'm a bit of a JS novice though, and wouldn't mind a second opinion if either one of you would be willing to give one.

The premise of removing Lodash in the first place is also not a sacred cow. It seems to be causing a few issues (#259) and we have such a minimal need for it that it seems reasonable to me to get rid of it, but let me know if you disagree here.

File: lib/utils.js [hunk 1]
```diff
@@ -1,25 +1,102 @@
...
 var hasOwn = {}.hasOwnProperty;
 var toString = {}.toString;
 
 var utils = module.exports = {
 
+  // Note that in ES6 all uses of this can just be replaced with Object.assign.
+  // Given a few more years we should drop this function completely.
+  assign: function(o1, o2) {
+    for (var attr in o2) {
+      o1[attr] = o2[attr];
```
Comment (asessa) [inline lib/utils.js]: if (typeof o2[attr] !== 'function')

**Comment by asessa** [comment]
Hey @brandur, I've found the problem and it's external to stripe-node. In my case Object is extended by the web framework.
As utils.assign is used for headers, I recommend to check also for content type to avoid conflicts.

File: lib/utils.js [hunk 2]
```diff
@@ -1,25 +1,102 @@
...
+    }
+    return o1;
+  },
```
Comment (SlexAxton) [inline lib/utils.js]: This is a pretty naive implementation. I'd feel much more comfortable bringing in something like the `object-assign` module to do just this. https://www.npmjs.com/package/object-assign

File: lib/utils.js [hunk 3]
```diff
@@ -1,25 +1,102 @@
...
   },
 
-  isObject: function(o) {
-    return _.isPlainObject(o);
+  /**
+   * Checks if `value` is a plain object, that is, an object created by the
+   * `Object` constructor or one with a `[[Prototype]]` of `null`.
+   *
+   * Note that this function is vendored from Lodash with some minor
+   * modifications.
```
Comment (SlexAxton) [inline lib/utils.js]: Seem again like we could just pull in https://www.npmjs.com/package/lodash.isplainobject and be safer.

**Comment by SlexAxton** [comment]
I don't feel suuuuuuuuuper strongly, but I don't think it's harmful to pull in dependencies like this, especially in the node world.

**Comment by floatingLomas** [comment]
Looks kool, but as an alternative, all the lodash methods are individually packaged, so we could just include [isPlainObject](https://www.npmjs.com/package/lodash.isplainobject) on its own.

**Comment by brandur** [comment]
@SlexAxton @floatingLomas Nice. Thanks a lot for the quick responses guys!

I think I've integrated all your feedback. I pulled out `lodash` in favor of `lodash.isplainobject` (which I didn't know about before — nice), and I've also put in the `object-assign` ponyfill for now.

It looks like @asessa's original problem may have been unrelated to lodash, but I feel like overall this change is still for the best: we get a path to a complete ES6 implementation instead of the relatively specialized `extend`, and we also just decrease the total dependency surface area.

r? @floatingLomas Would you mind taking a more complete review pass here if you have some time? Thanks!

**Comment by asessa** [comment]
Hey @brandur, latest commit using object-assign also solves my problem.

**Comment by brandur** [comment]
> Hey @brandur, latest commit using object-assign also solves my problem.

Excellent to hear! Thanks for following up.

**Comment by floatingLomas** [comment]
lgtm

I'm not 100% certain on the differences in behavior (if any) between `for (var k in data)` vs `_.forEach()`, but it's all just swapping ~equivalent functions, so I'm pretty sure it's cool.

**Comment by SlexAxton** [comment]
The only difference is the scope that the function runs in, which isn't an issue here. Also we're filtering on `hasOwnProperty` so it's all pretty safe/good.

**Comment by floatingLomas** [comment]
Awesomo. :+1:

**Comment by brandur** [comment]
@floatingLomas @SlexAxton Thanks again guys!

**Pull request merged**